Source/UI/Support/UserDefaultsExtension.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+private enum UserDefaultsKey: String, CaseIterable {
+    case hostname = "IPFTestHostname"
+    case port = "IPFTestPort"
+    case configType
+    case streams
+    case duration
+    case enableHaptics
+    case enableSounds
+}
+
+public extension UserDefaults {
```

Comment:
I would rather have our own configuration object that then knows how to persist itself than have a tight coupling on `UserDefaults` (and that class would just talk to `NSUserDefaults`). What if we want to move to something else in the future? Testing would also be simpler that way.

Reply:
How about we split this into another PR? I think the current PR as its own is huge.

Also this needs a little bit more hashing out. If we make the current IPFTestRunnerConfiguration into a swift Codable struct, we can get away with having to have tight coupling with user defaults and serialize and save the whole config in any way (JSON, Plist, UserDefaults, or something else). And even save the user's preference settings in there as those are also kind of a configuration for the test. But this will mean not using remodel. 

Let me know